Ase interface

YuuuuXie opened the pull request:
1. add *ase* folder, and remove `ase*.py` in *flare* folder
2. `ase` module includes flare calculator, otf engine and logger
3. different md engines are tested with otf (src: `otf_md.py`, unit test: `test_ase_setup/test_ase_otf.py` with QE)
4. add `.rst` for `mgp` and `ase` in *docs* folder
5. cleaned up `*.out` in *tests* folder since they are not used

jonpvandermause commented on flare/ase/calculator.py:
```diff
@@ -0,0 +1,166 @@
+        res.append(predict_on_atom((structure, atom, gp_model)))
+    return res
+
+def predict_on_structure_en(strucutre, gp_model):
```
Input should be "structure"

jonpvandermause commented on flare/ase/calculator.py:
```diff
@@ -0,0 +1,166 @@
+       
+        self.mgp_model = MappedGaussianProcess(self.gp_model, grid_params, struc_params)
+
+def predict_on_atom(params):
```
These functions look pretty similar to the ones in predict.py, would it be possible to use those so that we don't repeat code?

jonpvandermause commented on flare/mgp/otf.py:
```diff
@@ -18,7 +17,7 @@
 from flare import output
 
-class MFFOTF(OTF):
+class MGPOTF(OTF):
```
We now have three otf.py files in the repo -- mgp/otf, which inherits from flare/otf, and ase/otf, which is separate. If there are common methods to all three, we should try to import them from one place to minimize duplication.

YuuuuXie commented:
Jon: We now have three otf.py files in the repo -- mgp/otf, which inherits from flare/otf, and ase/otf, which is separate. If there are common methods to all three, we should try to import them from one place to minimize duplication.

YuuuuXie commented:
One common method that can be extracted out is `is_std_in_bound` which is sort of independent. MGPOTF is short because it inherits most of the methods from OTF. ase/otf is coupled with ase calculator, ase logger and ase md engines, so will be harder to be put out

jonpvandermause commented:
Yeah, is_std_in_bound would be good to have in one place (#30). We can either move it out of class in otf.py, or put it in predict.py.

YuuuuXie commented:
I can also have ase/otf to inherit from OTF somehow, since there're a lot of common attributes. Could we probably put `is_std_in_bound` in `util.py`? I think moving it in `predict.py` is a bit weird

jonpvandermause commented:
> Oh I used to use the same dir as you do (../test_files/pseudos) and never passed TRAVIS test. Then Lixin figured out TRAVIS did its unit test in the tests folder, so it runs python test_ase_setup/test_ase_otf.py. That's why I changed it to test_files/pseudos and then it passed unit test

Good catch. The test failed on my machine, but that's because I was running a different command than Travis. I switched it back.

anjohan commented:
Re: Travis testing directory.

I set Travis to run its tests from the tests directory. When I try to run pytest from the base directory, I get a lot of errors from trying to locate `./test_files/cp2k_input_1.in` etc.

YuuuuXie commented:
> Re: Travis testing directory.
> 
> I set Travis to run its tests from the tests directory. When I try to run pytest from the base directory, I get a lot of errors from trying to locate `./test_files/cp2k_input_1.in` etc.

I also tried running `pytest ./` in the *tests/* folder, but it works

YuuuuXie commented:
> Yeah, is_std_in_bound would be good to have in one place (#30). We can either move it out of class in otf.py, or put it in predict.py.

The latest commits removed the predict functions in ase/otf.py, and uses predict.py. Also removed `is_std_in_bound` in otf.py and ase/otf.py, and moved it to util.py 

This pull request can close #26 , #30 , #55
